cpp/zotaut/journal_timeliness_checker.cc

```diff
@@ -48,17 +48,20 @@ bool MaxDeliveredAtSmallerThanUpdateWindow(const time_t &max_delivered_at, const
 
 
 void ProcessJournal(ZoteroHarvester::Util::UploadTracker * const upload_tracker, const std::string &journal_name,
-                    const std::string &zeder_id, const std::string &zeder_instance, const unsigned update_window,
+                    const std::string &zeder_id, const std::string &zeder_instance,
+                    const ZoteroHarvester::Config::UploadOperation delivery_mode, const unsigned update_window,
                     std::string * tardy_list)
 {
-    // Make sure articles stored as online first are retried after an update_window period
-    upload_tracker->deleteOnlineFirstEntriesOlderThan(StringUtil::ToUnsigned(zeder_id), zeder_instance, update_window);
+     // Make sure articles stored as online first are retried after half an update_window period or at most 14 days
+    upload_tracker->deleteOnlineFirstEntriesOlderThan(StringUtil::ToUnsigned(zeder_id), zeder_instance, std::min(update_window / 2 , static_cast<unsigned>(14)));
 
-    const time_t max_delivered_at(upload_tracker->getLastUploadTime(StringUtil::ToUnsigned(zeder_id),
-                                  Zeder::ParseFlavour(zeder_instance)));
+    if (delivery_mode == ZoteroHarvester::Config::UploadOperation::LIVE) {
```

Comment:
As far as i can remember, one of the problems with the journal_timeliness_checker has been that both nu and ub28 process all journals. So e.g. nu might complain about missing articles for live journals, even if it is not his job to process live journals at all. Can we solve that somehow? And how do we test it? (maybe handle the delivery_mode depending on server environment or CLI parameter?)

=> might also be handled in a follow-up PR

Reply:
I thought about integrating some kind of host name determination and handling but for the time being removing overdue OF-articles seemed the decisive concern right now. Behaviour of the tardy journals should be identical previous versions